src/libfuncs/array.rs

```diff
@@ -1485,9 +1484,9 @@ pub fn build_tuple_from_span<'ctx, 'this>(
         CoreTypeConcrete::Array(info) => (&info.ty, registry.get_type(&info.ty)?),
         CoreTypeConcrete::Snapshot(info) => match registry.get_type(&info.ty)? {
             CoreTypeConcrete::Array(info) => (&info.ty, registry.get_type(&info.ty)?),
-            _ => unreachable!(),
+            _ => native_panic!("unexpected CoreTypeConcrete found"),
         },
-        _ => unreachable!(),
+        _ => native_panic!("unexpected CoreTypeConcrete found"),
```

Comment:
Could you give a little more information on why it's an unexpected type?

Reply:
Done!